Fixed the retrieval of the local LSP4IJ path.

anusreelakshmi934 opened the pull request:
Fixes #1242

vaisakhkannan commented on build.gradle:
```diff
@@ -148,10 +148,11 @@ dependencies {
-                'com.redhat.devtools.lsp4ij:' + lsp4ijVersion
-        plugins lsp4ij
+        def lsp4ijZip = file("../lsp4ij/build/distributions").listFiles()?.findAll { it.name.endsWith(".zip") }?.first()
+        providers.gradleProperty("useLocal").get() == 'true' ?
```
I am wondering whether the condition will always evaluate to false, because we have provided a false value [here](https://github.com/OpenLiberty/liberty-tools-intellij/blob/a967e46fafa0cb2dd234529188969452b1fd61dc/gradle.properties#L1).

Can you please confirm that ? Please share the Cron Job build links that you tested with this change

anusreelakshmi934 replied:
By default, the value is set to `false`. However, when we run the command with `-PuseLocal=true`, it changes to `true`, allowing us to use the local LSP4IJ. I verified this by running IDE locally using `./gradlew runIdeForUiTests -PuseLocal=true`, which installed LSP4IJ `0.10.0-SNAPSHOT`, the latest version from the main branch. Without `-PuseLocal=true`, it installed version `0.8.1`, as specified in `build.gradle`.

dessina-devasia commented on build.gradle:
```diff
@@ -148,10 +148,11 @@ dependencies {
-                file("../lsp4ij/build/distributions/LSP4IJ/") :
-                'com.redhat.devtools.lsp4ij:' + lsp4ijVersion
-        plugins lsp4ij
+        def lsp4ijZip = file("../lsp4ij/build/distributions").listFiles()?.findAll { it.name.endsWith(".zip") }?.first()
```
@anusreelakshmi934 Can you please check the code format here?

anusreelakshmi934 replied:
I have formatted the code directly in IntelliJ IDE, and this is the result it provided.

anusreelakshmi934 commented:
Cron Job link - https://github.com/anusreelakshmi934/liberty-tools-intellij/actions/runs/13025040119

dessina-devasia reviewed (approved):
Looks good to me. Thanks!

dessina-devasia commented on build.gradle:
```diff
@@ -148,10 +148,11 @@ dependencies {
-        def lsp4ij = providers.gradleProperty('useLocal') && providers.gradleProperty('useLocal') == 'true' ?
-                file("../lsp4ij/build/distributions/LSP4IJ/") :
-                'com.redhat.devtools.lsp4ij:' + lsp4ijVersion
-        plugins lsp4ij
```
@anusreelakshmi934 It will be better to add a comment corresponding to this change

anusreelakshmi934 replied:
I have added the comment

dessina-devasia reviewed (dismissed):
LGTM

mrglavas reviewed (changes requested):
@anusreelakshmi934 It occurred to me that there's an unzip step in our build: https://github.com/OpenLiberty/liberty-tools-intellij/blob/main/.github/workflows/build.yaml that seems no longer necessary with these changes. We should clean that up to avoid any confusion.

![image](https://github.com/user-attachments/assets/e6e0245c-2f8d-4a17-8ac2-03f6fef2194d)

anusreelakshmi934 commented:
> @anusreelakshmi934 It occurred to me that there's an unzip step in our build: https://github.com/OpenLiberty/liberty-tools-intellij/blob/main/.github/workflows/build.yaml that seems no longer necessary with these changes. We should clean that up to avoid any confusion.
> 
> ![image](https://private-user-images.githubusercontent.com/890521/407906197-e6e0245c-2f8d-4a17-8ac2-03f6fef2194d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgyMTE1MTYsIm5iZiI6MTczODIxMTIxNiwicGF0aCI6Ii84OTA1MjEvNDA3OTA2MTk3LWU2ZTAyNDVjLTJmOGQtNGExNy04YWMyLTAzZjZmZWYyMTk0ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEzMFQwNDI2NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OTFjZWE5MTk1M2VkNWUxZTYwNjZiNWQyZWY5MmQ0NWRjNjYzMjExYWFmMzkzZjE2YzVjYWNlOGVjYTliMTkwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.9-vZmFdd7u8bhAGkrIiuKnqn3JiNFTW5WwptK2WYI4c)

Yes @mrglavas . I will open a ticket for this and clean up the code. Thanks

mrglavas commented:
> > @anusreelakshmi934 It occurred to me that there's an unzip step in our build: https://github.com/OpenLiberty/liberty-tools-intellij/blob/main/.github/workflows/build.yaml that seems no longer necessary with these changes. We should clean that up to avoid any confusion.
> > ![image](https://private-user-images.githubusercontent.com/890521/407906197-e6e0245c-2f8d-4a17-8ac2-03f6fef2194d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgyMTE1MTYsIm5iZiI6MTczODIxMTIxNiwicGF0aCI6Ii84OTA1MjEvNDA3OTA2MTk3LWU2ZTAyNDVjLTJmOGQtNGExNy04YWMyLTAzZjZmZWYyMTk0ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEzMFQwNDI2NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OTFjZWE5MTk1M2VkNWUxZTYwNjZiNWQyZWY5MmQ0NWRjNjYzMjExYWFmMzkzZjE2YzVjYWNlOGVjYTliMTkwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.9-vZmFdd7u8bhAGkrIiuKnqn3JiNFTW5WwptK2WYI4c)
> 
> Yes @mrglavas . I have cleaned up the code. Here is the Cron Job link after the change: [link](https://github.com/anusreelakshmi934/liberty-tools-intellij/actions/runs/13045705589/job/36395825273).
> 
> In this run, the 24.0.9 build [fails](https://github.com/anusreelakshmi934/liberty-tools-intellij/actions/runs/13045705589/job/36395822047#step:7:33) because it lacks the unzip step, preventing it from retrieving the `lsp4ij` folder correctly. However, the 24.0.12 build succeeds since the condition is not functioning as expected. Should we proceed with this change despite the failure in 24.0.9?

@anusreelakshmi934 I believe in our team discussion earlier today we agreed that the unzip step should be restored to allow the 24.0.9 build to run. We also agreed that a comment to document that the unzip step is required for compatibility with the 24.0.9 build.

anusreelakshmi934 commented:
> > > @anusreelakshmi934 It occurred to me that there's an unzip step in our build: https://github.com/OpenLiberty/liberty-tools-intellij/blob/main/.github/workflows/build.yaml that seems no longer necessary with these changes. We should clean that up to avoid any confusion.
> > > ![image](https://private-user-images.githubusercontent.com/890521/407906197-e6e0245c-2f8d-4a17-8ac2-03f6fef2194d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgyMTE1MTYsIm5iZiI6MTczODIxMTIxNiwicGF0aCI6Ii84OTA1MjEvNDA3OTA2MTk3LWU2ZTAyNDVjLTJmOGQtNGExNy04YWMyLTAzZjZmZWYyMTk0ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEzMFQwNDI2NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OTFjZWE5MTk1M2VkNWUxZTYwNjZiNWQyZWY5MmQ0NWRjNjYzMjExYWFmMzkzZjE2YzVjYWNlOGVjYTliMTkwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.9-vZmFdd7u8bhAGkrIiuKnqn3JiNFTW5WwptK2WYI4c)
> > 
> > 
> > Yes @mrglavas . I have cleaned up the code. Here is the Cron Job link after the change: [link](https://github.com/anusreelakshmi934/liberty-tools-intellij/actions/runs/13045705589/job/36395825273).
> > In this run, the 24.0.9 build [fails](https://github.com/anusreelakshmi934/liberty-tools-intellij/actions/runs/13045705589/job/36395822047#step:7:33) because it lacks the unzip step, preventing it from retrieving the `lsp4ij` folder correctly. However, the 24.0.12 build succeeds since the condition is not functioning as expected. Should we proceed with this change despite the failure in 24.0.9?
> 
> @anusreelakshmi934 I believe in our team discussion earlier today we agreed that the unzip step should be restored to allow the 24.0.9 build to run. We also agreed that a comment be added to document that the unzip step is required for compatibility with the 24.0.9 build.

I have added a comment above the unzip step.

dessina-devasia reviewed (approved):
Looks good. Thanks!

vaisakhkannan reviewed (approved):
Looks good
